imagelab/ImgProvider.java

```diff
@@ -1,502 +1,610 @@
-            num = (num << 8) + black;
-            num = (num << 8) + black;
-            num = (num << 8) + black;
+            final int numOfValues = 3;
+            blueToBW = num & ALPHA;
+            num = num >> RANGE;
+            greenToBW = num & ALPHA;
+            num = num >> RANGE;
+            redToBW = num & ALPHA;
+            num = num >> RANGE;
+            alphaToBW = num & ALPHA;
+            blackToBW = (redToBW + greenToBW + blueToBW) / numOfValues;
+            num = alphaToBW;
+            num = (num << RANGE) + blackToBW;
+            num = (num << RANGE) + blackToBW;
+            num = (num << RANGE) + blackToBW;
             pix[i] = num;
         }
-        if (all) showPix("Black and White");
-        try { Thread.sleep(300);}catch(Exception e){}
-    }//toBW
-    
-    
-    /** Alias for showPix. (Syntactic sugar) */
-    public void showImage(String name) {
+        if (all) {
+            showPix("Black and White");
+        }
+        try {
+            Thread.sleep(SLEEP_TIME);
+        } catch (Exception e) {
+        }
+    }
+
+
+    /**
+     * Alias for showPix. (Syntactic sugar)
+     *
+     * @param name The title for the window.
+     */
+    public void showImage(final String name) {
         showPix(name);
-    }//showImage
-    
+    }
+
     /**
      * Display this image in a window.
+     *
      * @param name The title for the window.
      */
-    public void showPix(String name) {
+    public void showPix(final String name) {
         //System.out.println("ImgProvider:showPix:  before readIn");
-        if (!isLoaded) readinImage();
+        if (!isLoaded) {
+            readinImage();
+        }
         //System.out.println("ImgProvider:showPix:  after readIn");
         img = getToolkit().createImage(
                 new MemoryImageSource(pixwidth, pixheight, pix, 0, pixwidth));
         //System.out.println("ImgProvider:showPix:  before displayImage");
-        DisplayImage dis = new DisplayImage(this,name,true);
+        DisplayImage dis = new DisplayImage(this, name, true);
         //System.out.println("ImgProvider:showPix:  after displayImage");
-        try { Thread.sleep(100);}catch(Exception e){}       //make sure image has time to display
-    }//showPix
-        
+        try {
+            Thread.sleep(SLEEP_TIME);
+        } catch (Exception e) {
+        }
+    }
+
     /**
      * Pull the image apart into its RGB and Alpha components.
      */
     void separateColors() {
-        if (pix == null) return;
-        alpha   = new short[pixheight][pixwidth];
-        red     = new short[pixheight][pixwidth];
-        green   = new short[pixheight][pixwidth];
-        blue    = new short[pixheight][pixwidth];
+        if (pix == null) {
+            return;
+        }
+        alpha = new short[pixheight][pixwidth];
+        red = new short[pixheight][pixwidth];
+        green = new short[pixheight][pixwidth];
+        blue = new short[pixheight][pixwidth];
```

Comment:
Although this appears to adhere to convention, the rewrite seems less readable than the original.